### PR TITLE
Add HeaderFields.Len

### PR DESCRIPTION
**Pull request by foxcpp**
Separated from #75.

### CODE REVIEW

File: textproto/header.go [hunk 1]
```diff
@@ -282,9 +293,14 @@ func (fs *headerFieldsByKey) Del() {
...
 
+func (fs *headerFieldsByKey) Len() int {
+	return len(fs.h.m[fs.k])
+}
+
 // FieldsByKey iterates over all fields having the specified key.
 //
-// The header may not be mutated while iterating, except using HeaderFields.Del.
+// The header may not be mutated while iterating, except using HeaderFields.Del
+// and HeaderFields.Set.
```
Comment (emersion) [inline textproto/header.go:303]: This line should not be changed.

File: textproto/header.go [hunk 2]
```diff
@@ -220,9 +226,14 @@ func (fs *headerFields) Del() {
...
 
+func (fs *headerFields) Len() int {
+	return len(fs.h.l)
+}
+
 // Fields iterates over all the header fields.
 //
-// The header may not be mutated while iterating, except using HeaderFields.Del.
+// The header may not be mutated while iterating, except using HeaderFields.Del
+// and HeaderFields.Set.
```
Comment (emersion) [inline textproto/header.go:236]: This line should not be changed.

File: textproto/header.go (hunk 1 above)
Comment (foxcpp) [reply textproto/header.go:303]: woops, forgot to remove that one

**Comment by foxcpp** [comment]
git, what you have done!

**Pull request merged**